cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java

```diff
@@ -215,6 +215,12 @@ public Query query() {
         return query;
     }
 
+    /**
+     * <p>
+     * Closes the DocumentStore instance.
+     * </p>
+     *
```

Comment:
This may not be the correct place, but I wonder if we should add some documentation around the expected open/close behaviour. For example, I think the only way to safely access the `DocumentStore` is for the open/close to be controlled from something like an application lifecycle where an instance is obtained at the start and closed at the end.
If not I think we end up in situations like this:
Thread 1: `ds1 = DocumentStore.getInstance(...)`
Thread 2: `ds2 = DocumentStore.getInstance(...)`
ds1 and ds2 are the same instance, now if thread 1 calls `close()` accesses on thread 2 will get `IllegalStateException` for the `DocumentStore` being closed.